### PR TITLE
feat: introduce prefer-provide-intl-description rule

### PR DESCRIPTION
**Pull request by dkrutskikh**
### What is the purpose of this pull request? (put an "X" next to an item)

- [ ] Documentation update
- [ ] Bug fix
- [x] New rule
- [ ] Changes an existing rule
- [ ] Add autofixing to a rule
- [ ] Add a CLI option
- [ ] Add something to the core
- [ ] Other, please explain:

### CODE REVIEW

File: lib/src/analyzers/lint_analyzer/rules/rules_list/prefer_provide_intl_description/prefer_provide_intl_description_rule.dart [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import '../../../models/severity.dart';
+import '../../models/intl_rule.dart';
+import '../../rule_utils.dart';
+
+part 'visitor.dart';
+
+class PreferProvideIntlDescriptionRule extends IntlRule {
+  static const String ruleId = 'prefer-provide-intl-description';
+
+  static const _warning = 'Prefer provide description for translated message';
```
Comment (incendial) [inline lib/src/analyzers/lint_analyzer/rules/rules_list/prefer_provide_intl_description/prefer_provide_intl_description_rule.dart:19]: `'Provide description for translated message.';`

Comment (dkrutskikh) [reply lib/src/analyzers/lint_analyzer/rules/rules_list/prefer_provide_intl_description/prefer_provide_intl_description_rule.dart]: fixed

File: website/docs/rules/index.mdx [hunk 2]
```diff
@@ -712,6 +712,15 @@ Rules are grouped by category to help you understand their purpose. Each rule ha
   methods name argument.
 </RuleEntry>
 
+<RuleEntry
+  name="prefer-provide-intl-description"
+  type="intl"
+  severity="warning"
+  version="5.5.0"
+>
+  Warns when a `Intl` method invocation without description.
```
Comment (incendial) [inline website/docs/rules/index.mdx:721]: Warns when a method is invoked without a description.

File: website/docs/rules/intl/prefer-provide-intl-description.mdx [hunk 3]
```diff
@@ -0,0 +1,83 @@
+import RuleDetails from '@site/src/components/RuleDetails';
+
+<RuleDetails version="5.5.0" severity="warning" />
+
+Warns when a `Intl.message()`, `Intl.plural()`, `Intl.gender()` or `Intl.select()` method invocate without description.
```
Comment (incendial) [inline website/docs/rules/intl/prefer-provide-intl-description.mdx:5]: Warns when the .... methods are invoked without a description.

Comment (dkrutskikh) [reply website/docs/rules/intl/prefer-provide-intl-description.mdx]: fixed

File: website/docs/rules/index.mdx (hunk 2 above)
Comment (dkrutskikh) [reply website/docs/rules/index.mdx]: fixed

**Review by incendial — APPROVED** [review]

**Pull request merged**